lib/licensee/projects/github_project.rb

```diff
@@ -0,0 +1,43 @@
+# Github project
+#
+# Analyses a remote Github repository for license information
+#
+# Only the root folder of a repository will be scanned for performance reasons!
+
+require 'octokit'
+
+module Licensee
+  module Projects
+    class GithubProject < Licensee::Projects::Project
+      # If there's any trailing data (e.g. `.git`) this pattern will ignore it:
+      # we're going to use the API rather than clone the repo.
+      GITHUB_REPO_PATTERN = %r{https://github.com/([^\/]+\/[^\/\.]+).*}
+
+      def initialize(github_url, **args)
+        @repo_name = github_url[GITHUB_REPO_PATTERN, 1]
+        raise ArgumentError, "Not a github URL: #{github_url}" unless @repo_name
+        super(**args)
+      end
+
+      private
+
+      def files
+        @files ||= fetch_files.map do |data|
+          {
+            name:         data[:name],
+            contents_url: data[:download_url],
+            dir:          '/'
+          }
+        end
+      end
+
+      def load_file(file)
+        Octokit.get URI(file[:contents_url])
+      end
+
+      def fetch_files
+        Octokit.contents(@repo_name).select { |data| data[:type] == 'file' }
```

Comment:
Does `repo` make more sense than `repo_name`? Whatever we call it, I think we can use `attr_accessor` to mirror the other project classes's style.